### DISCUSSION TITLE
How to reference element by name?

### DISCUSSION THREAD

**Discussion by isaacn-lab** [datapane/datapane#172] [Help] [answered]
Hello, 

First of all, thank you for datapane. I am a new to it, but I find it very useful and easy to start working with.

Each class has optional name argument [(A unique name for the block to reference when adding text or embedding)](https://datapane.github.io/datapane/report/blocks.html#datapane.client.api.report.blocks.Text), is it possible to use it to create a link to the element in the report (I couldn't find any examples)?

Thank you!
-Nikita

**Comment by mands** [comment, marked as answer]
Hi @ZKpot ,

Apologies for the delay in replying. Yes it's possible to use the name as a link, right now you have to create the link manually, by adding `?block_name={NAME}` to the end of the report url, where `{NAME}` is the name you gave to the block. 

e.g. the demo report at https://datapane.com/u/datapane/reports/q34yQE7/demo-report/ has a block called `table1`, I can reference that block using the following URL - https://datapane.com/u/datapane/reports/q34yQE7/demo-report/?block_name=table1
